### ISSUE TITLE
Legacy Boot Code Not Loaded in v0.13.3.AD Pre-Release

### ISSUE THREAD

**Issue by tjthrush** [dakanji/RefindPlus#155]
### RefindPlus Version

Pre-Release Code Build

### Device Type

Apple Mac

### Problem Description

using MyBootMgr v088, no choice is available to choose a valid BC disk. Works properly in v087

### Problem Point

Before starting a loader or tool

### Affected Items

Win OS Loader

### Debug Log

[23d22w5046_v087.log](https://github.com/dakanji/RefindPlus/files/10829116/23d22w5046_v087.log)
[23d23y5701_v088.log](https://github.com/dakanji/RefindPlus/files/10829117/23d23y5701_v088.log)

**Comment by dakanji** [comment]
Thanks. Gives a period to look at.
Will temporarily put this on hold while trying to look into the other more serious issue of RefindPlus hanging.

**Comment by dakanji** [comment]
This one will need a few tests to find the breaking commit.
Please start with X277

[X277-BOOTx64.zip](https://github.com/dakanji/RefindPlus/files/10829852/X277-BOOTx64.zip)

**Comment by tjthrush** [comment]
Joy!
1st try, you are **talented**-
[23d25v2745.log](https://github.com/dakanji/RefindPlus/files/10832124/23d25v2745.log)

**Comment by dakanji** [comment]
Unfortunately, not quite!

Just going through a Git Bisect process and all that gave was that the item had not yet been broken as at the commit the build was done on. 

So, we have to move up the commit history a bit and test at another point:
[X278-BC03-BOOTx64.zip](https://github.com/dakanji/RefindPlus/files/10832207/X278-BC03-BOOTx64.zip)

If this works, we move up again. If it fails, we try the halfway point  and cycle until the breaking commit is found.
You can read up on Git Bisect on the web to understand the process.

**Comment by tjthrush** [comment]
no joy, blank screen is back
[23d25w2422.log](https://github.com/dakanji/RefindPlus/files/10832241/23d25w2422.log)

**Comment by dakanji** [comment]
I see. Please try this instead.
[X279-BC04-BOOTx64.zip](https://github.com/dakanji/RefindPlus/files/10832251/X279-BC04-BOOTx64.zip)

**Comment by tjthrush** [comment]
to clarify, I deleted the X278-BC03- prefix prior to running

**Comment by tjthrush** [comment]
no joy (04), blank screen gone, boots but **no BC choice**
[23d25w5013.log](https://github.com/dakanji/RefindPlus/files/10832283/23d25w5013.log)

**Comment by dakanji** [comment]
That's fine. Please try this.
[X280-BC05-BOOTx64.zip](https://github.com/dakanji/RefindPlus/files/10832309/X280-BC05-BOOTx64.zip)

**Comment by tjthrush** [comment]
no joy (05), blank screen gone, boots but no BC choice.
Ran this twice. Doesn't seem to be updating logs.
Populating desktop with items took 75+ sec in Big Sur.

**Comment by dakanji** [comment]
Thanks. Please try the attached two to close this aspect out.
[X281-BOOTx64.zip](https://github.com/dakanji/RefindPlus/files/10832930/X281-BOOTx64.zip)

**Comment by tjthrush** [comment]
no joy (06), blank screen gone, boots but no BC choice.

Populating desktop with items nominal time in Big Sur.
[23d26f3631.log](https://github.com/dakanji/RefindPlus/files/10833057/23d26f3631.log)

running 07 next

**Comment by tjthrush** [comment]
no joy (07), blank screen gone, boots but no BC choice.

Populating desktop with items took 80+ sec in Big Sur.
[23d26f5016.log](https://github.com/dakanji/RefindPlus/files/10833077/23d26f5016.log)

**Comment by dakanji** [comment]
The "All over the place" outcomes, along with the hang from before and manifesting on one unit but not another, point at a memory conflict issue. These are a PITA to nail down but let's see how far we get with a process I have used before.

Can you please run the attached pair and let me know how they do?
[X281-BOOTx64.zip](https://github.com/dakanji/RefindPlus/files/10833974/X281-BOOTx64.zip)


Thanks

**Comment by tjthrush** [comment]
Glad to help troubleshoot these nasty bugs. Didn't realize the degree of sensitivity in memory conflicts developers need to work around.

no joy (B14), blank screen gone, boots but no BC choice.

Populating desktop with items nominal time in Big Sur (faster than Mojave).

[23d26t3812.log](https://github.com/dakanji/RefindPlus/files/10834333/23d26t3812.log)

running T14 next

**Comment by tjthrush** [comment]
no joy (T14), blank screen gone, boots but no BC choice.

Populating desktop with items nominal time in Big Sur (faster than Mojave) but didn't resolve thumbnail images for another 50+ s. Mojave has no delay.


[23d26t5707.log](https://github.com/dakanji/RefindPlus/files/10834387/23d26t5707.log)

**Comment by dakanji** [comment]
OK. Those were a dead end. The memory conflict is not getting triggered on your machine any longer. I'll have to wait until I stumble on the cause or it manifests on mine or someone else's.

Let's try to see whether we can get more info on the BootCamp thing with X282. Rename to the usual name.
[X282-BOOTx64-NPT.zip](https://github.com/dakanji/RefindPlus/files/10834433/X282-BOOTx64-NPT.zip)

NB: This outputs a detailed log and is therefore significantly slower to load.

**Comment by tjthrush** [comment]
no joy (NPT), blank screen gone, boots but no BC choice.

Populating desktop with items nominal time in Big Sur (faster than Mojave) no delay resolving thumbnail images.

[23d26v0234.log](https://github.com/dakanji/RefindPlus/files/10834529/23d26v0234.log)

**Comment by dakanji** [comment]
X283 should work. Please confirm.
[X283-BOOTx64.zip](https://github.com/dakanji/RefindPlus/files/10834586/X283-BOOTx64.zip)

**Comment by tjthrush** [comment]
some joy :)
BC choice and successful boot into win7
[23d26w0708.log](https://github.com/dakanji/RefindPlus/files/10834687/23d26w0708.log)

However, boots into Big Sur populating desktop with items 60+ s. Also notice finder issues, window movement, apple menu lack of stickiness, 'Favorites' items in window side-bar not blue.

[23d26w2123.log](https://github.com/dakanji/RefindPlus/files/10834700/23d26w2123.log)

Getting closer? Sorry-

**Comment by dakanji** [comment]
Sorry, but what happens in MacOS is not controlled by RefindPlus.

**Comment by tjthrush** [comment]
Mojave is fine. 
'Favorites' items in window side-bar not blue is a Mojave 'feature' and a key diff Apple rolled into Big Sur (perhaps Catalina as well?) Ran this twice after BC/Win7 boot. Would you like me to boot controlled by RefindPlus again?

**Comment by dakanji** [comment]
See my previous

**Comment by tjthrush** [comment]
> See my previous

??? Running Mojave is bypassing RefindPlus on my 5,1. Perhaps I'm not understanding your comment

**Comment by dakanji** [comment]
What I am saying is that whatever you are seeing in MacOS has no relationship to RefindPlus.
I suppose I should have mentioned this when you were giving feedback earlier on the time it takes for load desktop etc.

In summary, there is nothing further to test on this.
The separate memory conflict thing (already tagged as a known issue) will come out in the wash sooner or later.

Thanks for the multiple test runs. Much appreciated.
